qa/tasks/cephfs/test_acls.py

```diff
@@ -0,0 +1,480 @@
+                                  acl.find(self.currentuser) == -1 and
+                                 acl.find(self.currentgroup) == -1 and
+                                 acl.find('mask') == -1]
+        self.removable_acls = [acl for acl in self.all_acls if
+                               acl.find(self.currentuser) != -1 or
+                               acl.find(self.currentgroup) != -1 or
+                               acl.find('mask') != -1]
+
+        self.testdir1 =  os.path.join(self.mount_a.mountpoint, "dir1")
+        self.testdir2 = os.path.join(self.testdir1, 'dir2')
+        self.testdir3 = os.path.join(self.testdir2, 'dir3')
+        self.testdirs = [self.testdir1, self.testdir2, self.testdir3]
+
+        self.__setup_once_executed_once = True
+
+        # TODO: test without passwordless sudo
+        # self.mount_a.run_as_root(['useradd', '-s', '/bin/bash', self.otheruser])
+        # self.mount_a.run_shell(['sudo', 'passwd', self.otheruser, '--stdin'], stdin=self.otheruserpwd)
+
+    def setup(self, mkdir=True, change_perms=True, change_grp_perms=True):
+        self.setup_once()
+        self.fs.set_ceph_conf('client', 'fuse_default_permissions', False)
+        self.fs.set_ceph_conf('client', 'client_acl_type', 'posix_acl')
+        # XXX: above options are effective only for new mounts, so remounting
+        # mount a.
+        self.mount_a.umount_wait()
+        self.mount_a.mount()
+        self.mount_a.wait_until_mounted()
+
+        if mkdir:
+            self.create_test_dirs(change_perms=change_perms,
+                                  change_grp_perms=change_grp_perms)
+
+    def get_corresponding_rm_acl(self, acl):
+        # if ACL is removable
+        if acl.find(self.currentuser) != -1 or \
+           acl.find(self.currentgroup) != -1 or acl.find('mask') != -1:
+            return acl[ : acl.find('rwx') ]
+        else:
+            return None
+
+    def create_test_dirs(self, perm_bits=None, as_currentuser=False,
+                         change_perms=True, change_grp_perms=True):
+        if perm_bits is None:
+            cmd = 'mkdir --parents %s' % (self.testdir3)
+        else:
+            cmd = 'mkdir --parents --mode=%s %s' % (perm_bits, self.testdir3)
+
+        if as_currentuser:
+            self.mount_a.run_shell(cmd)
+        else:
+            self.mount_a.run_as_root(cmd)
+
+        if change_perms:
+            if change_grp_perms:
+                self.mount_a.run_as_root('chmod --recursive g+rwx ' + self.testdir1)
+            self.mount_a.run_as_root('chmod --recursive o-rwx ' + self.testdir1)
+
+    def return_list(self, var):
+        if str(type(var)).find('list') != -1:
+            return var
+        elif str(type(var)).find('str') != -1:
+            return [var]
+        else:
+            raise RuntimeError('var "%s" (of type %s) should be either list or str.' % (var, str(type(var))))
+
+    def assert_acls_are_(self, state, acls, testdirs):
```

Comment:
nit: trailing underscore on the name here is odd, maybe remove it?

Reply:
The trailing underscore makes sense when reading calls made to this method. Calls can be `assert_acl_are('present', ...` or `assert_acl_are('absent', ...`. Just `assert_acls_are` sounded alot incomplete to me.

If you think this doesn't make sense, we can drop it.

Reply:
Replaced the method name in upcoming push by `assert_acls_are()` and added `assert_acls_are_present()` and `assert_acls_are_absent()`. This should it make it more convenient to write.